vm/src/vm.h

```diff
@@ -80,6 +83,114 @@ size_t moshell_vm_next_page(moshell_vm vm);
  */
 void moshell_vm_free(moshell_vm vm);
 
+/**
+ * A moshell value, either an i64, a unsigned byte, a double
+ * or an address
+ * */
+typedef union {
+    const int64_t i;
+    const uint8_t b;
+    const double d;
+    const void *const ptr;
+} moshell_value;
+
+/**
+ * Different kinds of vm objects
+ * */
+typedef enum {
+    OBJ_STR,
+    OBJ_INT,
+    OBJ_DOUBLE,
+    OBJ_VEC
+} moshell_object_type;
+
+/**
+ * A moshell heap object, with its type
+ * */
+typedef struct {
+    moshell_object_type type;
+    const void *val;
+} moshell_object;
+
+/**
+ * A sized array of values
+ * */
+typedef struct {
+    const uint64_t size;
```

Comment:
```suggestion
    size_t size;
```
I would drop the `const` qualifier here, as the view does not impact the actual vector, and to be in sync with the other structs that don't have it. Anyway, the main thing here is that as the vector implementation depend on the host (the size cannot be greater than the size of a pointer), using platform dependent types here make sense.

Reply:
it feels wrong that the array's size can change but not its pointer, i added more consts to other structs to be more in sync